src/github.com/matrix-org/dendrite/federationapi/writers/threepid.go

```diff
@@ -120,11 +121,30 @@ func createInviteFrom3PIDInvite(
 	}
 
 	if !queryRes.RoomExists {
-		// TODO: Use federation to auth the event
-		return nil, nil
+		// Use federation to auth the event
+		var remoteServers []gomatrixserverlib.ServerName
+		_, remoteServers[0], err = gomatrixserverlib.SplitID('@', inv.Sender)
```

Comment:
This will have problems because the remoteServers slice will have length 0.
```go
remoteServers := make([]gomatrixserverlib.ServerName, 2)
```
might be better here.

Reply:
Indeed. Fixed in https://github.com/matrix-org/dendrite/pull/222/commits/ebe91ac251f4f963625cf2fd3d9142d95407cc30